tests/test_s3_to_json.py

```diff
@@ -83,117 +83,309 @@ def json_file_basenames_dict(self):
         }
         return json_file_basenames
 
-    def test_write_healthkitv2samples_file_to_json_dataset(self, s3_obj, namespace, monkeypatch):
-        monkeypatch.setattr("boto3.client", lambda x: MockAWSClient())
+    def test_transform_object_to_array_of_objects(self):
+        json_obj_to_replace = {
+                "0": 60.0,
+                "1": 61.2,
+                "2": "99"
+        }
+        transformed_object = s3_to_json.transform_object_to_array_of_objects(
+                json_obj_to_replace=json_obj_to_replace,
+                key_name="key",
+                key_type=int,
+                value_name="value",
+                value_type=int
+        )
+        expected_object = [
+                {
+                    "key": 0,
+                    "value": 60
+                },
+                {
+                    "key": 1,
+                    "value": 61
+                },
+                {
+                    "key":2,
+                    "value": 99
+                }
+        ]
+        assert all([obj in expected_object for obj in transformed_object])
+
+    def test_transform_json_generic(self):
+        sample_metadata = {
+                "type": "FitbitDevices",
+                "start_date": datetime.datetime(2022, 1, 12, 0, 0),
+                "end_date": datetime.datetime(2023, 1, 14, 0, 0),
+        }
+        transformed_json = s3_to_json.transform_json(
+                json_obj={},
+                dataset_identifier=sample_metadata["type"],
+                metadata=sample_metadata
+        )
+
+        assert (
+            sample_metadata["start_date"].isoformat()
```

Comment:
You may want to explicitly write out the isoformat(). This is because if ever (isoformat) has a bug, this test would pass.  (Highly unlikely but just something to think about when writing tests that use the same functions as the code you're testing.

Reply:
That's an interesting edge case that `isoformat` might have a bug, although it's from the python standard library. I don't think it's an edge case we need to test for. That would be up to python's developers to test for bugs in `datetime`. 

Perhaps a more likely scenario is that the default ISO format outputted by `isoformat` changes. I would be okay with that, as long as it's ISO standard.

Reply:
Makes sense to me!